internal/monitor/monitor.go

```diff
@@ -0,0 +1,219 @@
+package monitor
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"os"
+	"path/filepath"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/api/types/mount"
+	"github.com/docker/docker/api/types/network"
+	"github.com/docker/go-connections/nat"
+	"github.com/pkg/errors"
+	"go.uber.org/zap"
+
+	"github.com/spinup-host/spinup/config"
+	"github.com/spinup-host/spinup/internal/dockerservice"
+	"github.com/spinup-host/spinup/misc"
+)
+
+const (
+	PrometheusContainerName = "spinup-prometheus"
+	PgExporterContainerName = "spinup-postgres-exporter"
+	NetworkName             = "spinup_services"
+)
+
+// Target represents a postgres service for monitoring.
+// The assumption is that all services are running on the same host, hence only ports should differ
```

Comment:
meaning the `UserName` and `Password` also should be the same across Postgres services?

Reply:
Oh no, I should clarify the comment then. Everything in the `Target` struct can actually be different. Host is the same so it's not in the struct.